ipatests/test_integration/test_backup_and_restore.py

```diff
@@ -861,3 +874,43 @@ def test_replica_install_after_restore(self):
         # install second replica after restore
         tasks.install_replica(master, replica2)
         check_replication(master, replica2, "testuser2")
+
+
+class TestBackupAndRestoreTrust(IntegrationTest):
+    """Test for Backup and Restore for Trust scenario"""
+    topology = 'star'
+
+    def test_restore_trust_pkg_before_restore(self):
+        """Test restore with adtrust when trust-ad pkg is missing.
+
+        Test for bug https://pagure.io/freeipa/issue/7630.
+        """
+        self.master.run_command(['yum', '-y',
+                                 'install', '*ipa-server-trust-ad'])
+        self.master.run_command(['ipa-adtrust-install', '-U',
+                                 '--enable-compat', '--netbios-name', 'IPA',
+                                 '-a', self.master.config.admin_password,
+                                 '--add-sids'])
+
+        with restore_checker(self.master):
+            backup_path = backup(self.master)
+
+            self.master.run_command(['ipa-server-install',
+                                     '--uninstall',
+                                     '-U'])
+
+            self.master.run_command(['yum', '-y',
+                                     'remove', '*ipa-server-trust-ad'])
+            tasks.install_master(self.master, setup_dns=True)
+
+            dirman_password = self.master.config.dirman_password
+            result = self.master.run_command(
+                ['ipa-restore', backup_path],
+                stdin_text=dirman_password + '\nyes',
+                raiseonerr=False)
+            assert 'Please install the package' in result.stderr_text
+
+            self.master.run_command(['yum', '-y',
```

Comment:
At this point ipa-restore failed, the local DNS is probably not running and I guess the machine's /etc/resolv.conf is still using localhost as DNS. You can run the test with an external DNS (do not add --setup-dns when configuring the server).